proto/spacemesh/v1/app_types.proto

```diff
@@ -0,0 +1,124 @@
+    AccountId author = 3; // template uploader
+
+    // The layer can be obtained from the tx receipt but it is convenient to have it here
+    LayerNumber deploy_layer = 4; // layer that the template was deployed - 0 for genesis templates
+
+    TransactionId deploy_tx_id = 5; // the id of the deploy template transaction. Empty for genesis templates.
+
+    bytes code = 6; // template wasm code
+
+    repeated VariableMetadata vars = 7; // template variables order and metadata
+}
+
+// A Spacemesh app
+// Note that both layer and spawner can be obtained from the transaction and transaction id for apps spawned by a transaction
+// but is convenient to return them directly as part of apps' metadata.
+message App {
+    AccountId address = 1; // App's address in global space
+    string name = 2; // short descriptive name
+    AppTemplateAddress template_address = 3; // App's template address
+    LayerNumber spawn_layer = 4; // Layer in which app was spawned
+    TransactionId spawn_tx_id = 5; // Transaction which spawned the app. empty for genesis apps.
+    AccountId spawner = 6; // App spawner account
+    repeated Variable variables = 8; // ordered array of app vars and their values
+}
+
+// A request to return a template by address from the template global state store
+message GetAppTemplateRequest {
+    AppTemplateAddress address = 1; // template address
+    bool include_code = 2; // when true, template result will include full wasm code
+}
+
+// A response with AppTemplate data
+message GetAppTemplateResponse {
+    AppTemplate template = 1;
+}
+
+// A request to return all templates from the templates global state store with pagination
+message GetAllAppTemplatesRequest {
+    uint32 max_results = 1; // max number of results to return
+    uint32 offset = 2; // query offset
+    bool include_code = 3; // when try templates results will include full wasm code
+}
+
+// A response includes total templates available in store and a list of zero ore more templates
+message GetAllAppTemplatesResponse {
+    uint32 total_results = 1; // total number of templates
+    repeated AppTemplate templates = 2;
+}
+
+// A request to get an app's variable by var id
+message GetAppVariableRequest {
+    AccountId address = 1; // app's address
+    LayerNumber layer_number = 2; // requested value as of a layer. Pass MAX_UI32 to get latest value
+    uint32 id = 3; // app's var id - var's unique identifier
+}
+
+// A response with variable meta-data and value
+message GetAppVariableResponse {
+    Variable variable_value = 1;
+}
+
+// A request to get a deployed app metadata and optionally all variable values as of a specific layer or latest state.
+message GetAppRequest {
+    AccountId address = 1; // app address
+    LayerNumber layer_number = 2; // layer for state. Ignored when include_var_values is set to false. pass MAX_UI32 to get latest state.
+    bool include_var_values = 3; // when true, var values will be included in response, false otherwise.
```

Comment:
I didn't add that option under the issue - but I do think it makes sense to have it